### PR TITLE
Pen testing environment changes

### PR DESCRIPTION
**Pull request by VenkataChalla554**
### Context
Creating pen testing environment
### Changes proposed in this pull request
Create pen testing.
### Guidance to review
verify : http://pen.register-trainee-teachers.service.gov.uk/
### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by VenkataChalla554** [comment]
[Register: pen test environment](https://trello.com/c/RWFaWoJ9/1705-register-pen-test-environment)

File: config/settings/pen.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# The url for this app, also used by `dfe_sign_in`
+base_url: https://sandbox.register-trainee-teachers.service.gov.uk
```
Comment (tomtrentham) [inline config/settings/pen.yml:2]: I think this needs to match the URL of the pen env, so maybe:
```suggestion
base_url: https://pen.register-trainee-teachers.service.gov.uk
```

File: config/settings/pen.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    school_direct_salaried: true
+    school_direct_tuition_fee: true
+    iqts: true
+  integrate_with_dqt: true
+  register_api: true
+  google:
+    send_data_to_big_query: false
+
+environment:
+  name: sandbox
```
Comment (tomtrentham) [inline config/settings/pen.yml:41]:
```suggestion
  name: pen
```

**Review by tomtrentham — COMMENTED** [review]
A couple of tweaks but otherwise looks good.

File: config/settings/pen.yml (hunk 1 above)
Comment (VenkataChalla554) [reply config/settings/pen.yml]: Fixed

File: Makefile [hunk 3]
```diff
@@ -79,6 +79,12 @@ staging:
 	$(eval DTTP_HOSTNAME=traineeteacherportal-pp)
 	$(eval export TF_VARS=-var config_short=${CONFIG_SHORT} -var service_short=${SERVICE_SHORT} -var service_name=${SERVICE_NAME} -var azure_resource_prefix=${RESOURCE_NAME_PREFIX})
 
+pen:
+	$(eval include global_config/pen.sh)
+	$(eval DEPLOY_ENV=pen)
+	$(eval DTTP_HOSTNAME=traineeteacherportal-pen)
```
Comment (saliceti) [inline Makefile:85]: DTTP_HOSTNAME can be removed (and from the other make targets as well)

File: config/settings/pen.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ dfe_sign_in:
   profile: https://profile.signin.education.gov.uk
 
 recruits_api:
-  base_url: "https://sandbox.apply-for-teacher-training.service.gov.uk/register-api"
+  base_url: "https://pen.apply-for-teacher-training.service.gov.uk/register-api"
```
Comment (tomtrentham) [inline config/settings/pen.yml:14]: Is there a `pen` environment for the Apply service (`apply-for-teacher-training.service.gov.uk`)?

**Review by tomtrentham — COMMENTED** [review]
Looking good, just one question on the URL for the Apply API.

File: config/settings/pen.yml (hunk 4 above)
Comment (VenkataChalla554) [reply config/settings/pen.yml:14]: Hi Tom, there is no pentesting environment for apply-for-teaching , I will fix this by reverting .

Comment (VenkataChalla554) [reply config/settings/pen.yml]: fixed

**Review by defong — APPROVED** [review]
Can login in to http://pen.register-trainee-teachers.service.gov.uk/ and there is no pii

File: terraform/custom_domains/environment_domains/workspace_variables/register_pen.tfvars.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      "front_door_name": "s189p01-rtt-edu-domains-fd",
+      "resource_group_name": "s189p01-rttdomains-rg",
+      "domains": [
+        "pen"
+      ],
+      "cached_paths": [
+        "/packs/*"
+      ],
+      "environment_short": "pen",
+      "origin_hostname": "pen.test.teacherservices.cloud",
```
Comment (RMcVelia) [inline terraform/custom_domains/environment_domains/workspace_variables/register_pen.tfvars.json:19]: i would have thought register-pen.test.teacherservices.cloud

File: terraform/custom_domains/environment_domains/workspace_variables/register_pen.tfvars.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      "front_door_name": "s189p01-rtt-svc-domains-fd",
+      "resource_group_name": "s189p01-rttdomains-rg",
+      "domains": [
+        "pen"
+      ],
+      "cached_paths": [
+        "/packs/*"
+      ],
+      "environment_short": "pen",
+      "origin_hostname": "pen.test.teacherservices.cloud",
```
Comment (RMcVelia) [inline terraform/custom_domains/environment_domains/workspace_variables/register_pen.tfvars.json:32]: register-pen.test.teacherservices.cloud

File: terraform/aks/workspace-variables/pen.tfvars.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
+{
+  "env_config": "pen",
+  "app_environment": "pen",
+  "key_vault_name": "s189t01-rtt-pen-kv",
+  "key_vault_app_secret_name": "REGISTER-APP-SECRETS-PEN",
+  "key_vault_infra_secret_name": "BAT-INFRA-SECRETS-PEN",
```
Comment (RMcVelia) [inline terraform/aks/workspace-variables/pen.tfvars.json:6]: There are -STAGING secrets in the pen-kv, which I would think should be removed as they are overriding the -PEN secrets. The module loads all secrets from the key vault.

File: terraform/custom_domains/environment_domains/workspace_variables/register_pen.tfvars.json (hunk 5 above)
Comment (VenkataChalla554) [reply terraform/custom_domains/environment_domains/workspace_variables/register_pen.tfvars.json]: fixed and applied changes

File: terraform/custom_domains/environment_domains/workspace_variables/register_pen.tfvars.json (hunk 6 above)
Comment (VenkataChalla554) [reply terraform/custom_domains/environment_domains/workspace_variables/register_pen.tfvars.json]: fixed and applied changes

File: terraform/aks/workspace-variables/pen.tfvars.json (hunk 7 above)
Comment (VenkataChalla554) [reply terraform/aks/workspace-variables/pen.tfvars.json:6]: removed staging secret and updated the  APPLICATION_HOST .

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**